Feature: print QR options

andi34 opened the pull request:
<!--
    Thank you for contributing!
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)
<!-- Example:
- [x] Bug fix
-->

- [x] New feature

<!--
    Please ensure your pull request is ready:

    - Please run 'yarn build' before submitting to have consistent formatting for both JavaScript & SCSS
    - Please run 'yarn eslint' to make sure there's no lint issues
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

New settings while printing QR-Code onto the image:
- QR size adjustable
- QR offset adjustable
- QR position adjustable
- QR error correction adjustable

By default place the QR-Code on the bottom right corner, use an small QR-Code with an small offset of 10px to the edges.


#### Is there anything you'd like reviewers to focus on?
![Screenshot_20220125-232740](https://user-images.githubusercontent.com/6080900/151070345-f3c1c4a0-fc7b-4b46-9d64-efbea872c516.png)


![Screenshot_20220125-232752](https://user-images.githubusercontent.com/6080900/151070351-45f74de3-050d-4233-9f5d-7017c0cf1ff6.png)


![Screenshot_20220125-232758](https://user-images.githubusercontent.com/6080900/151070352-2497e7cc-b02f-426c-ba77-d1ef7457d15b.png)

andi34 committed:
feature(qr): place QR code onto the picture at print

New settings from print:
- QR size adjustable
- QR offset adjustable
- QR position adjustable

By default place the QR-Code on the bottom right corner,
use an small QR-Code with an small offset of 10px to the edges.

Change-Id: I28baec6a6d0903cb47e3bf4145a751510740d3c0

andi34 committed:
api(print): check if QR-Code offset is numeric

Change-Id: Iefa6f4666eaa337866ea0a6be8b8c96b6bade8eb

andi34 commented on api/print.php:
```diff
@@ -74,27 +82,52 @@
+                $image = imagecreatefrompng($filename_codes);
+                $resultRotated = imagerotate($image, 90, 0); // Rotate image
+                imagepng($resultRotated, $filename_codes, 0);
+                imagedestroy($image); // Destroy the created collage in memory
```
Not a collage, fix the comment

andi34 commented on api/print.php:
```diff
@@ -74,27 +82,52 @@
             }
             include '../vendor/phpqrcode/lib/full/qrlib.php';
-            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, 10);
+            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, $config['print']['qrSize']);
```
Make the ECC adjustable too?

up-87 replied:
Depends on how much impact it has. I didn't play around with it by now.

andi34 replied:
Not much, another switch case would need to be added. There's 4 ECC options we could use. Plus the config stuff for the Adminpanel.

up-87 replied:
In this case I'd stay away for it. But maybe choose a different level anyway? https://www.qrcode.com/en/about/error_correction.html
> Level Q or H may be selected for factory environment where QR Code get dirty

I don't feel like this is the typical use case or requirement for a photobooth...
https://blog.qrstuff.com/2011/12/14/qr-code-error-correction

> Level L or Level M represent the best compromise between density and ruggedness for general marketing use.

andi34 replied:
There might be different use cases which why I'd consider to add a switch and let the user decide, but sure we can get down to a lower profile by default while on it.

andi34 replied:
Might be enough to set $config['print']['qrEcc'] and make it a select list inside Adminpanel, so no need for a switch case

andi34 replied:
Or maybe in general for QR and move this setting to QR section. Same applies to the qr inside gallery

andi34 reviewed (commented):
@up-87 what do you think?

andi34 committed:
api(qr): make QR-Code error correction level adjustable

Lower default used error correction level from H to M.

Information:
https: //www.qrcode.com/en/about/error_correction.html
https: //blog.qrstuff.com/2011/12/14/qr-code-error-correction
Change-Id: I4849a340caeb4dfbc5933a90de4585ed8d932961

andi34 committed:
lang(en): write QR Code always the same

Change-Id: Iddde9a9258b56cb017ae247d22a65aec5602cc74
